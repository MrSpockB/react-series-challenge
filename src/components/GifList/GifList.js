import React, { Component } from 'react';

export class GifList extends Component {
  render() {
    return (
      <ul>
        <li>Gif</li>
        <li>Gif</li>
        <li>Gif</li>
      </ul>
    )
  }
}