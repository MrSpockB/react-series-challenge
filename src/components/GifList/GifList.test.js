import React from 'react';
import { shallow, render } from 'enzyme';
import { shallowToJson, renderToJson } from 'enzyme-to-json';
import { GifList } from "./GifList";
import {MemoryRouter} from "react-router-dom";

describe('Gif list component', () => {
  it('should render correctly', () => {
    const output = shallow(
      <GifList
        gifs={[]}
        addGif={jest.fn}
        removeGif={jest.fn}
        favoritesGifs={{allGifs: {}}}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should mark one gif as favorited', () => {
    const gifs = [{id: '123', title: 'test'}];
    const favoritedGifs = {'123': {id: '123', title: 'test'}};
    const output = render(
      <MemoryRouter>
        <GifList
          gifs={gifs}
          addGif={jest.fn}
          removeGif={jest.fn}
          favoritesGifs={{allGifs: favoritedGifs}}
        />
      </MemoryRouter>
    );
    expect(renderToJson(output)).toMatchSnapshot();
  });
});