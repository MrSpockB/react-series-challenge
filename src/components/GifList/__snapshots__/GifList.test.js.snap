// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif list component should mark one gif as favorited 1`] = `
<div
  class="sc-bZQynM UAeAy"
>
  <div
    class="sc-bdVaJa kiwtFJ"
  >
    <div
      class="sc-htpNat dGyvgD"
    >
      <img
        alt="test"
        class="sc-ifAKCX fPKiWZ"
        src=""
      >
        
      </img>
      <div
        class="sc-bwzfXH heRxYP"
      >
        <span
          class="sc-EHOje gwryyh"
        >
          Title: test
        </span>
        <button>
          Remove from favorites
        </button>
        <a
          class="sc-bxivhb iFAgbD"
          href="/detail/123"
        >
          View detail
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Gif list component should render correctly 1`] = `<styled.div />`;
